package/kedro_viz/server.py

```diff
@@ -474,7 +483,40 @@ def format_pipeline_data(
 @app.route("/api/nodes.json")
 def nodes_json():
     """Serve the pipeline data."""
-    return jsonify(data)
+    return jsonify(_DATA)
+
+
+@app.route("/api/nodes/<string:node_id>")
+def nodes_metadata(node_id):
+    """Serve the metadata for node and dataset."""
+    node = _NODES.get(node_id)
+    if node and node.get("type") == "task":
+        task_metadata = {
+            # pylint: disable=protected-access
+            "code": inspect.getsource(node["obj"]._func)
+        }
+        # pylint: disable=protected-access
+        code_full_path = Path(inspect.getfile(node["obj"]._func)).expanduser().resolve()
+
+        code_location = Path(code_full_path).relative_to(Path.cwd().parent)
```

Comment:
I used `Path.cwd()` to get the project root, given that this endpoint is only accessible when users run `kedro viz` from the project root (this endpoint is not reachable if they do `kedro viz --load-file=data.json`). 

Another way to get the project root is to store context object in a global `_CONTEXT` variable (or store a dictionary if they use Kedro<0.15.* since context was introduced in 0.15.0). If we prefer this approach, I can change that too :)

Reply:
Why is it relative to the _parent_ of the current directory?

Reply:
Also, `code_full_path` is an instance of `Path` already, no need to wrap it the second time.

Reply:
For simplicity, can't we just return `code_full_path`? If not, maybe it's worth putting this github comment into the code as a comment.

Reply:
This is to address https://github.com/quantumblacklabs/kedro-viz/pull/231#pullrequestreview-461701527. 
For example, 
Code full path would be `/Users/kiyohito_kunii/Desktop/hello/src/hello/pipelines/data_engineering/nodes.py`

and cwd is `/Users/kiyohito_kunii/Desktop/hello/src/hello/` (so the parent of this is `/Users/kiyohito_kunii/Desktop/hello/src/`) 
but we want to only add  `hello/src/hello/pipelines/data_engineering/nodes.py` in JSON,

Reply:
@limdauto I used to return the full path, but @richardwestenra also had a point about this in https://github.com/quantumblacklabs/kedro-viz/pull/231#pullrequestreview-461701527 

We have to agree on either format: either simple code full path or only showing from the project root (e.g `hello/src/hello/pipelines/data_engineering/nodes.py`).